easybuild/easyconfigs/d/DGL/DGL-0.9.1-foss-2021a-CUDA-11.3.1.eb

```diff
@@ -0,0 +1,135 @@
+easyblock = 'CMakeMake'
+
+name = 'DGL'
+version = '0.9.1'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://www.dgl.ai'
+description = """DGL is an easy-to-use, high performance and scalable Python package for deep learning on graphs.
+DGL is framework agnostic, meaning if a deep graph model is a component of an end-to-end application, the rest
+of the logics can be implemented in any major frameworks, such as PyTorch, Apache MXNet or TensorFlow."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+github_account = 'dmlc'
+source_urls = [GITHUB_LOWER_SOURCE]
+sources = [
+    {
+        'download_filename': '%(version)s.tar.gz',
+        'filename': '%(namelower)s-%(version)s.tar.gz',
+    },
+    {
+        'source_urls': ['https://github.com/KarypisLab/METIS/archive'],
+        'download_filename': 'v5.1.1-DistDGL-v0.5.tar.gz',
+        'filename': 'metis-5.1.1-DistDGL-v0.5.tar.gz',
+        'extract_cmd': "tar -C %(namelower)s-%(version)s/third_party/METIS --strip-components=1 -xf %s",
+    },
+    {
+        'source_urls': ['https://github.com/KarypisLab/GKlib/archive'],
+        'download_filename': 'METIS-v5.1.1-DistDGL-0.5.tar.gz',
+        'filename': 'GKlib-METIS-v5.1.1-DistDGL-0.5.tar.gz',
+        'extract_cmd': "tar -C %(namelower)s-%(version)s/third_party/METIS/GKlib --strip-components=1 -xf %s",
+    },
+    {
+        'filename': 'tensorpipe-20230206.tar.gz',
+        'git_config': {
+            'url': 'https://github.com/pytorch',
+            'repo_name': 'tensorpipe',
+            'commit': '6042f1a4cbce8eef997f11ed0012de137b317361',
+            'recursive': True,
+        },
+        'extract_cmd': "tar -C %(namelower)s-%(version)s/third_party/tensorpipe --strip-components=1 -xf %s",
+    },
+    # DGL really needs cub >= 1.17, CUDA 11.3 only have 1.11
+    {
+        'source_urls': ['https://github.com/NVIDIA/thrust/archive'],
+        'download_filename': '1.17.0.tar.gz',
+        'filename': 'thrust-1.17.0.tar.gz',
+        'extract_cmd': "tar -C %(namelower)s-%(version)s/third_party/thrust --strip-components=1 -xf %s",
+    },
+    {
+        'source_urls': ['https://github.com/NVIDIA/cub/archive'],
+        'download_filename': '1.17.0.tar.gz',
+        'filename': 'cub-1.17.0.tar.gz',
+        'extract_cmd':
+            "tar -C %(namelower)s-%(version)s/third_party/thrust/dependencies/cub --strip-components=1 -xf %s",
+    },
+]
+patches = [
+    '%(name)s-%(version)s_use_externals_instead_of_submodules.patch',
+]
+checksums = [
+    {'dgl-0.9.1.tar.gz': '8d26ebb7ed976665bbf5bbd1792d8e6efb13a8fa16e5eb1efed75e07fb982e04'},
+    {'metis-5.1.1-DistDGL-v0.5.tar.gz': 'cedf0b32d32a8496bac7eb078b2b8260fb00ddb8d50c27e4082968a01bc33331'},
+    {'GKlib-METIS-v5.1.1-DistDGL-0.5.tar.gz': '52aa0d383d42360f4faa0ae9537ba2ca348eeab4db5f2dfd6343192d0ff4b833'},
+    {'tensorpipe-20230206.tar.gz': 'a4246afac190ffc1bef438920cea490a9aa8dfc5983e765b420ec4408331618e'},
```

Comment:
checksum can't be used here. Pretty sure this dict syntax doesn't allow for `None`  values either last i tried. So, probably need a framework fix for that, ugh

Reply:
ah forgot about that one...